Userland/Applications/PixelPaint/ImageEditor.cpp

```diff
@@ -383,6 +385,21 @@ void ImageEditor::mousemove_event(GUI::MouseEvent& event)
     auto image_event = event_with_pan_and_scale_applied(event);
     if (on_image_mouse_position_change) {
         on_image_mouse_position_change(image_event.position());
+
+        // Update the cursors when the mouse position is changed to allow the rectangles and ellipses to be clamped within
+        // the bounds of the ImageEditor and not draw ontop of the rest of the UI.
+        if (is<BrushTool>(*m_active_tool)) {
+            if (m_active_tool->tool_name() == "Erase Tool"sv) {
+                EraseTool* active_erase_tool = dynamic_cast<EraseTool*>(m_active_tool);
+                active_erase_tool->set_current_position(m_mouse_position);
+                active_erase_tool->refresh_editor_cursor();
+            }
+            if (m_active_tool->tool_name() == "Brush Tool"sv) {
+                BrushTool* active_erase_tool = dynamic_cast<BrushTool*>(m_active_tool);
+                active_erase_tool->set_current_position(m_mouse_position);
+                active_erase_tool->refresh_editor_cursor();
+            }
```

Comment:
This looks odd, there's two cases here but they both seem the same? Why aren't these just virtual methods on the tool? Making this just:
```
m_active_tool->set_current_position(m_mouse_position);
m_active_tool->refresh_editor_cursor();
```
(You could make the default implementation just a stub)

Reply:
Other brushtools don't have this implementation at all. If we make the default then the others just override that implementation? i.e. brush and erase tool which actually use these.

Reply:
Just this to the base tool:
```
virtual void set_current_position(Gfx::IntPoint) const {}
virtual void refresh_editor_cursor() {}
```
Then only the tools that actually use that need to override/implement it.